crates/hir-ty/src/infer/unify.rs

```diff
@@ -916,6 +916,31 @@ impl<'a> InferenceTable<'a> {
 
     /// Check if given type is `Sized` or not
     pub(crate) fn is_sized(&mut self, ty: &Ty) -> bool {
+        let mut ty = ty.clone();
+        {
+            let mut structs = SmallVec::<[_; 8]>::new();
+            // Must use a loop here and not recursion because otherwise users will conduct completely
+            // artificial examples of structs that have themselves as the tail field and complain r-a crashes.
+            while let Some((AdtId::StructId(id), subst)) = ty.as_adt() {
+                let struct_data = self.db.struct_data(id);
+                if let Some((last_field, _)) = struct_data.variant_data.fields().iter().last() {
```

Comment:
Should we do `struct_data.variant_data.fields().iter().rev().next()` here? It's `DoubleEndedIterator`, so it might be a bit faster.

Reply:
I didn't touch this part but yeah I guess. Do you want me to change this in this PR?

Reply:
Sure, let's do that if it works. LTGM otherwise.

Reply:
Done.

Reply:
I think you have a stray semicolon.